test/calendar_test.js

```diff
@@ -0,0 +1,288 @@
+var expect = require('chai').expect;
+var Calendar = require('../src/data/calendar');
+
+const {getFrenchBankHolidays, getBoxingDays} = require('../src/data/french-bank-holidays');
+global.getFrenchBankHolidays = getFrenchBankHolidays;
+global.getBoxingDays = getBoxingDays;
+
+
+describe("Calendar", () => {
+  describe("#monthWeekTable()", () => {
+    let weektable;
+
+    beforeEach(() => {
+      weekTable = Calendar.monthWeekTable(2017, 1);
+    });
+
+    it("should return a week array with the expected length", () => {
+      expect(weekTable)
+        .to.have.length.of(6);
+    });
+
+    it("should return a week array with the expected weeks length", () => {
+      expect(weekTable.every(week => week.length === 7))
+        .eql(true);
+    });
+
+    it("should return a week array with the null day type when not part of the month", () => {
+      weekTable.forEach(days => {
+        // ensure every day of each week has a null type whenever the day is not
+        // part of the target month
+        days.forEach(day => {
+          if (day.date.getMonth() !== 0) {
+            expect(day.type).to.be.a("null");
+          } else if (day.date.getDay() === 0) {
+            expect(day.type).eql("WE");
+          } else {
+            expect(day.type).to.be.a("string");
+          }
+        })
+      });
+    });
+  });
+
+  describe('#guessTypeFromComment', () => {
+    it("should return CP for an empty comment.", () => {
+      var type = Calendar.guessTypeFromComment("");
+      expect(type).eql("CP");
+    });
+
+    it("should return CP by default.", () => {
+      var type = Calendar.guessTypeFromComment("Sailing of the Britany coast.");
+      expect(type).eql("CP");
+    });
+
+    it("should return CP for a comment containing CP.", () => {
+      var type = Calendar.guessTypeFromComment("1 CP");
+      expect(type).eql("CP");
+    });
+
+    it("should return JRTT for a comment containing RTT.", () => {
+      var type = Calendar.guessTypeFromComment("RTT");
+      expect(type).eql("JRTT");
+    });
+
+    it("should return CS for a comment containing CSS.", () => {
+      var type = Calendar.guessTypeFromComment("CSS");
+      expect(type).eql("CS");
+    });
```

Comment:
Apparently everything that is not JF, CP or JRTT is a CS.
We test it because it is part of the current spec, if the spec changes we should change the test but right now that the behavior we are aiming for.

it is not coming from nowhere I talked about it with @sylvestre